app/src/main/java/com/example/househomey/HomeSelectStateFragment.java

```diff
@@ -0,0 +1,138 @@
+package com.example.househomey;
+
+import static com.example.househomey.utils.FragmentUtils.navigateToFragmentPage;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.Firebase;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.WriteBatch;
+
+import java.util.ArrayList;
+
+/**
+ * This fragment is a child of the home fragment containing the list of the user's inventory
+ * This fragment represents a state where items are selectable
+ * @see HomeFragment
+ * @author Sami Jagirdar
+ */
+public class HomeSelectStateFragment extends HomeFragment implements DeleteItemsFragment.DeleteCallBack{
+
+    private ItemAdapter itemView;
+
+    /**
+     * This constructs a new HomeFragment with the appropriate list of items
+     *
+     * @param itemRef A reference to the firestore collection containing the items to display
+     */
+    public HomeSelectStateFragment(CollectionReference itemRef) {
+        super(itemRef);
+    }
+
+    /**
+     *
+     * @param inflater The LayoutInflater object that can be used to inflate
+     * any views in the fragment,
+     * @param container If non-null, this is the parent view that the fragment's
+     * UI should be attached to.  The fragment should not add the view itself,
+     * but this can be used to generate the LayoutParams of the view.
+     * @param savedInstanceState If non-null, this fragment is being re-constructed
+     * from a previous saved state as given here.
+     *
+     * @return the home fragment view containing the inventory list
+     * in its base state where items are selectable
+     */
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        // Inflate the fragment's layout
+        View rootView = inflater.inflate(R.layout.fragment_home_select, container, false);
+
+        itemListView = rootView.findViewById(R.id.item_list);
+        itemAdapter = new ItemAdapter(getContext(), itemList);
+        itemListView.setAdapter(itemAdapter);
+        itemView = (ItemAdapter) itemAdapter;
+        itemView.setSelectState(true);
+        itemRef.addSnapshotListener(this::setupItemListener);
```

Comment:
By my other comment, we would delete this line

Reply:
Sounds good, so just remove the listener when we are in SelectState yes?

Reply:
btw, I did make deletion be committed in a batch, so we don't have to worry about the listener listening during the process of deletion.

Reply:
Yeah, batch looks beautiful. I think it just simplifies everything moving the listener. And yeah just the two things, remove the listener and send back to base state after deletion

Reply:
Ok, so tried removing the listener, but turns out that whenever we navigate to Select state, since it's a new fragment being initialized, the items disappear as there's no listener to populate the newly initialized item list.

So I can do 2 things here. Either I keep the listener in select state or I make it so that whenever I navigate from base state to select state, in my bundling, I pass the populated item list to select state. Which do you prefer?

Reply:
I think bundling and sending the data makes the most sense